src/main/java/io/github/nucleuspowered/nucleus/modules/warn/commands/CheckWarningsCommand.java

```diff
@@ -0,0 +1,84 @@
+import io.github.nucleuspowered.nucleus.api.data.WarnData;
+import io.github.nucleuspowered.nucleus.internal.annotations.*;
+import io.github.nucleuspowered.nucleus.internal.command.CommandBase;
+import io.github.nucleuspowered.nucleus.internal.permissions.SuggestedLevel;
+import io.github.nucleuspowered.nucleus.modules.warn.handlers.WarnHandler;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.CommandElement;
+import org.spongepowered.api.command.args.GenericArguments;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.service.user.UserStorageService;
+import org.spongepowered.api.text.Text;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Checks the warnings of a player.
+ *
+ * Command Usage: /checkwarnings user Permission: quickstart.checkwarnings.base
+ */
+@Permissions(suggestedLevel = SuggestedLevel.MOD)
+@RunAsync
+@NoWarmup
+@NoCooldown
+@NoCost
+@RegisterCommand({"checkwarnings", "checkwarn", "warnings"})
+public class CheckWarningsCommand extends CommandBase<CommandSource> {
+
+    @Inject private WarnHandler handler;
+    private final String playerKey = "player";
+
+    @Override
+    public CommandElement[] getArguments() {
+        return new CommandElement[] {GenericArguments.onlyOne(GenericArguments.user(Text.of(playerKey)))};
+    }
+
+    @Override
+    public CommandResult executeCommand(CommandSource src, CommandContext args) throws Exception {
+        User user = args.<User>getOne(playerKey).get();
+
+        List<WarnData> warnings = handler.getWarnings(user);
+        if (warnings.isEmpty()) {
+            src.sendMessage(Util.getTextMessageWithFormat("command.checkwarnings.none", user.getName()));
+            return CommandResult.success();
+        }
+
+        int index = 0;
+        for (WarnData warning : warnings) {
+            String name;
+            if (warning.getWarner().equals(Util.consoleFakeUUID)) {
+                name = Sponge.getServer().getConsole().getName();
+            } else {
+                Optional<User> ou = Sponge.getServiceManager().provideUnchecked(UserStorageService.class).get(warning.getWarner());
+                name = ou.isPresent() ? ou.get().getName() : Util.getMessageWithFormat("standard.unknown");
+            }
+
+            String time = "";
+            if (warning.getEndTimestamp().isPresent()) {
+                time = Util.getTimeStringFromSeconds(Instant.now().until(warning.getEndTimestamp().get(), ChronoUnit.SECONDS));
+            } else if (warning.getTimeFromNextLogin().isPresent()) {
+                time = Util.getTimeStringFromSeconds(warning.getTimeFromNextLogin().get().getSeconds());
+            } else {
+                time = "the rest of time";
```

Comment:
Move this into a resource string in `message.properties` to make it accessible to translators.